### PR TITLE
add grass-dev

### PR DESCRIPTION
**Pull request by ninsbl**
This PR adds a new grass-dev package that could be used for e.g nightly or weekly builds of the development version.
The code is based on the grass package.  It is basically tested (meaning grass starts and runs a couple of tests succesfully).
Please review critically, esp. with regards to naming of the package, executable or app directory. @landam and @hellik, please have a look as well. I am open to any suggestion for changes. I would like to use the build recipe in CI. Therfore I added a CI variable to make it run as a github action. I hope that is acceptable...

The contained patch of the GRASS source I hope we can inetgrate upstreams at some point...

Let me know if you need something tested specifically, before integration/packaging...

Would be great to have a GRASS 8 preview packaged in the course of FOSS4G I would say...

### CODE REVIEW

File: src/grass-dev/osgeo4w/package.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		mingw-w64-x86_64-fftw \
+		mingw-w64-x86_64-lapack \
+		mingw-w64-x86_64-cairo
+	"
+	cmd.exe /c "$cmd" || cmd.exe /c "$cmd" || cmd.exe /c "$cmd"
+
+	cd ../$P-$V
+
+if [ "$CI" ] ; then
+	P=$(cygpath -ua "C:/Program Files (x86)\Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64/")
```
Comment (jef-n) [inline src/grass-dev/osgeo4w/package.sh:85]: P is supposed to contain the package name.  What's wrong with following

```
        export VCPATH=$(
                vs2019env
                echo ${PATH//\/cygdrive/}
        )
```

File: src/grass-dev/osgeo4w/package.sh [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	sed -i "3 a export PATH=\"$P:$PATH\"" mswindows/osgeo4w/mklibs.sh
+	sed -i "s/dumpbin -exports/dumpbin \/EXPORTS/" mswindows/osgeo4w/mklibs.sh
```
Comment (jef-n) [inline src/grass-dev/osgeo4w/package.sh:87]: update the patch instead?

File: src/grass-dev/osgeo4w/package.sh [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+export PACKAGE_POSTFIX="-dev"
+export V=8.0.dev
+export B=next
+export MAINTAINER=JuergenFischer
+export BUILDDEPENDS="gdal-devel pdal-devel proj-devel geos-devel netcdf-devel libjpeg-devel libpng-devel libpq-devel libtiff-devel sqlite3-devel zstd-devel python3-ply python3-core python3-six python3-pywin32 python3-wxpython wxwidgets-devel"
+
+branch=main
+
+if [ "$CI" ] ; then
+cd "$OSGEO4W_REP"
```
Comment (jef-n) [inline src/grass-dev/osgeo4w/package.sh:11]: OSGEO4W_REP is supposed to be the directory where the package repository is (which is not neccessarilly the the base repository)

File: src/grass-dev/osgeo4w/package.sh (hunk 2 above)
Comment (ninsbl) [reply src/grass-dev/osgeo4w/package.sh:87]: Good point. Will update accordingly (also the variable name).

File: src/grass-dev/osgeo4w/package.sh (hunk 1 above)
Comment (ninsbl) [reply src/grass-dev/osgeo4w/package.sh:85]: thanks for your swift reply. For some reason the location of dumpbin did not get added to the path by vs2019env. Can have a second look at what actually happens...

File: src/grass-dev/osgeo4w/package.sh (hunk 3 above)
Comment (ninsbl) [reply src/grass-dev/osgeo4w/package.sh:11]: OK, I just use that as entry point in my github action where OSGEO4W_REP is set. Can use another variable name here.

**Comment by ninsbl** [comment]
BTW, i tried to add PDAL support, that would be really nice to have. But because that is compiled with MSVC I really struggle to make that work. Any suggestion on how (or if) that can be achieved in a msys2 build / or how it should work would be very much appreciated!

**Comment by jef-n** [comment]
v.in.pdal uses C++ API.  That won't work with different compilers involved.

**Comment by hellik** [comment]
https://pdal.io/CAPI/doxygen/html/index.html

**Comment by jef-n** [comment]
> https://pdal.io/CAPI/doxygen/html/index.html

Are you saying GRASS only used the C API?

https://github.com/OSGeo/grass/blob/f6487aafe2c046a3049d36e5107fd0b209114dd4/vector/v.in.pdal/main.cpp#L400-L401

looks like C++.

**Comment by hellik** [comment]
No, it's just the link to the C API for reference.

Not that GRASS is using it.

**Comment by hellik** [comment]
> No, it's just the link to the C API for reference.
> 
> Not that GRASS is using it.

@wenzeslaus any idea?

File: src/grass-dev/osgeo4w/package.sh (hunk 3 above)
Comment (ninsbl) [reply src/grass-dev/osgeo4w/package.sh]: Variable renamed in 3e8c9a8

File: src/grass-dev/osgeo4w/package.sh (hunk 2 above)
Comment (ninsbl) [reply src/grass-dev/osgeo4w/package.sh]: Moved into the package.sh patch in 3e8c9a8

File: src/grass-dev/osgeo4w/package.sh (hunk 1 above)
Comment (ninsbl) [reply src/grass-dev/osgeo4w/package.sh]: Still not 100% sure what happens here, maybe the PATH variable gets overwritten y subsequent calls of fetchenv...

**Comment by ninsbl** [comment]
> v.in.pdal uses C++ API. That won't work with different compilers involved.

OK, does that mean we either have to build GRASS with MSVC on Windows or build a (probably internal) version of pdal with g++?

Could using clang be an option that attempts to be compatible with both MSVC and GCC:
https://clang.llvm.org/docs/MSVCCompatibility.html

Or is any of thos options so unrealistic / unreliable that we rather have to accept the lack of PDAL support on MS Windows?

**Comment by wenzeslaus** [comment]
My intention was to use the C API, but at that time it was not really maintained (I don't know what is the situation now) and the recommendation I got from PDAL project people was to use C++ API, so that's what is used for r.in.pdal and v.in.pdal.

**Comment by ninsbl** [comment]
Based on the changes here I made a PR upstreams (https://github.com/OSGeo/grass/pull/1921) that would make the patch obsolet.

**Comment by ninsbl** [comment]
Depends on: https://github.com/OSGeo/grass/pull/1921

**Comment by ninsbl** [comment]
> Or is any of thos options so unrealistic / unreliable that we rather have to accept the lack of PDAL support on MS Windows?


It seems the only viable option to get PDAL support for now is to build GRASS with MSVC. There are some early atempts:
https://github.com/OSGeo/grass/pull/289
https://github.com/OSGeo/grass/pull/348
But unfortunately, unfinished (it seems to be a bigger change)...

So here, we leave PDAL out for now...

**Comment by ninsbl** [comment]
A gentle ping: RC2 was realeased for GRASS 8.0: https://github.com/OSGeo/grass/releases/tag/8.0.0RC2

It would be therefore really great to have a `stable`, `ltr` and `development` version for GRASS in OSGeo4W. Anything I or he GRASS GIS team could do to facilitate that?

Fully understand and appreciate that it needs to be included in a way that does not increase maintenance burden unreasonably...

**Pull request closed without merging**